server/src/main/java/io/crate/protocols/postgres/PostgresWireProtocol.java

```diff
@@ -451,8 +473,11 @@ private void finishAuthentication(Channel channel) {
             User authenticatedUser = authContext.authenticate();
             String database = properties.getProperty("database");
             session = sqlOperations.createSession(database, authenticatedUser);
+            activeSessions.put(this.keyData, session);
             Messages.sendAuthenticationOK(channel)
-                .addListener(f -> sendParamsAndRdyForQuery(channel));
+                .addListener(f -> sendParams(channel))
+                .addListener(f -> Messages.sendKeyData(channel, keyData.pid, keyData.secretKey))
```

Comment:
After the pg-tunnel branch is merged and once you rebase this you'll have to update the new `PgClient` to handle the key data: https://github.com/crate/crate/blob/cf5b7e6d1a5f2306fb9b7e2d5bd691196f20baa5/server/src/main/java/io/crate/protocols/postgres/PgClient.java#L352-L358

Reply:
I just handled `KeyData` [22f4dd4](https://github.com/crate/crate/pull/12208/commits/22f4dd4f00d9757ef7971f6240ed3412a4f9a9e5).
Does pg-tunnel need to be able to cancel?